File: android/app/src/main/java/sol/vortexx/android/net/api/AuthTokenSource.kt
package sol.vortexx.android.net.api

/**
 * Who owns the current JWT (access + refresh).
 *
 * The net layer only needs to *read* them; the auth feature (Wave 5)
 * writes after login / refresh. Keeping the interface read-only from the
 * HTTP side removes a whole class of accidental-mutation bugs where a
 * request-failure handler could overwrite the session.
 */
interface AuthTokenSource {
    /** Current access JWT, or null if not logged in. */
    suspend fun accessToken(): String?

    /** Refresh JWT used by the refresh endpoint; null if not available. */
    suspend fun refreshToken(): String?

    /** Called by net layer on 401 to attempt a silent refresh. */
    suspend fun refresh(): Boolean
}

File: ios/Modules/Sources/Net/api/AuthTokenSource.swift
import Foundation

/// Read-only view over the current JWT pair.
///
/// Net layer never mutates the session — only Auth does. Splitting the
/// interface keeps accidental overwrite paths impossible: an HTTP 401
/// handler can *ask* for a refresh but not invent one.
public protocol AuthTokenSource: Sendable {
    func accessToken() async -> String?
    func refreshToken() async -> String?
    /// Called on 401 — returns whether a fresh access token is now available.
    func refresh() async -> Bool
}
